src/utils/item.ts

```diff
@@ -130,3 +130,87 @@ export const paginationContentFilter = (
   items
     .filter((i) => i.type !== ItemType.FOLDER)
     .filter((i) => !i.settings?.isPinned);
+
+interface Node {
+  children: string[];
+}
+
+interface Tree {
+  [nodeId: string]: Node;
+}
+
+const createMapTree = (data: DiscriminatedItem[]) => {
+  const childrenTreeMap: Tree = {};
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < data.length; i++) {
+    const ele = data[i];
+    const parents = getParentsIdsFromPath(ele.path, { ignoreSelf: true });
+    const lastParent = parents[parents.length - 1];
+    if (lastParent) {
+      if (childrenTreeMap[lastParent]) {
+        childrenTreeMap[lastParent].children = [
+          ...childrenTreeMap[lastParent].children,
+          ele.id,
+        ];
+      } else {
+        childrenTreeMap[lastParent] = { children: [ele.id] };
```

Comment:
I don't think you need an object `{ children }`, why don't you use a plain array?